text/1003-cli-naming.rst

```diff
@@ -0,0 +1,258 @@
+
+1. With more and more functionality exposed to the CLI, the
+   ``<action>-<category>`` approach has the potential to balloon the output
+   of ``edgedb -h`` to more than 100 lines, making it effectively unreadable.
+
+   The ``<group> <command>`` scheme would only display the few top-level
+   categories making it easier for users to build a mental model of what's
+   possible with the CLI.
+
+2. The ``edgedb show-status`` command does not indicate what kind of status
+   it will show. It could be showing status of the current migration, or of
+   the currently running server, or of how many active connections the server
+   has at the moment.  Renaming it to ``edgedb show-migration-status`` would
+   make it longer to type and overall very verbose.
+
+   With ``<group> <command>`` scheme the solution is obvious:
+   ``edgedb migration status``.
+
+3. The ``edgedb show-status`` command prompts another question: what
+   if another kind of "display status" command is needed?
+
+   With ``<group> <command>`` scheme, there would be no conflict between
+   ``edgedb server status`` and ``edgedb migration status``.
+
+4. The ``<action>-<category>`` scheme does not allow category-specific
+   help sections.
+
+   With ``<group> <command>`` scheme, top-level help sections are natural:
+   ``edgedb migration -h`` command can have a description of the recommended
+   migrations workflow, while ``edgedb server -h`` would explain to the user
+   how to work with EdgeDB servers.
+
+5. ``edgedb configure`` currently implements the ``<group> <command>`` scheme;
+   it should have been ``edgedb insert-config``, ``edgedb set-config``,
+   ``edgedb reset-config``.
+
+6. The ``edgedb server`` CLI design outlined in RFC 1001 directly conflicts
+   with the ``<action>-<category>`` scheme.
+
+The last observation is what prompted the creation of this RFC.  We either
+need to update RFC 1001 to use the ``<action>-<category>`` scheme or
+adopt the ``<group> <command>`` for all other commands.  Combining two
+drastically different CLI designs in one tool is not an option.
+
+
+Overview
+========
+
+The RFC proposes to adopt the ``<group> <command>`` naming scheme for
+the ``edgedb`` CLI.  Here's an outline of current and future ``edgedb``
+subcommands:
+
+* ``server`` (see also RFC 1001.)
+  - ``init``
+  - ``install``
+  - ``restart``
+  - ``start``
+  - ``status``
+  - ``stop``
+
+* ``dump``
+  - ``db`` -- backup a database.
+  - ``all`` -- backup all databases, as well as roles, configs, etc into
+    a directory.
+  - ``restore-db``
+  - ``restore-all``
+  - ``config`` -- backup system configuration.
+
+* ``migration``
+  - ``status``
+  - ``create``
+  - ``apply``
+
+* ``role``
+  - ``create [--superuser]``
+  - ``alter``
+  - ``drop``
+  - ``list``
+
+* ``db``
+  - ``create``
+  - ``rename``
+  - ``drop``
+  - ``list``
+
+* ``config [--system]`` (used to be ``edgedb configure``)
+  - ``set``
+  - ``reset``
+  - ``add`` (used to be ``insert``)
```

Comment:
Can you elaborate why you think breaking consistency with the server command here is an improvement?

Reply:
Hm, in what way is the consistency broken here?